app/views/layouts/application.html.erb

```diff
@@ -8,7 +8,14 @@
 
 </head>
 <body>
-  <%= yield %>
```

Comment:
The only content that will show up for `yield(:title)` is content that is wrapped around:

```ruby
# one liner
content_for(:title, 'This is my title')

# multi-line version
content_for :title do
  # content here
end
```

or

```ruby
# one liner
provide(:title, 'This is my title')

# multi-line version
provide :title do
  # content
end
```

But for example, look at any view file like `app/views/sessions/index.html.erb`. `provide(:title, "Sessions")` will set the title just fine, but nothing else from that file will show up! You can see if you visit `/sessions`. There is no content other than the things you used `provide` for. There are no sessions listed, no link to create a new one, or any of the other content in that file.

That's what a plain `yield` with no other arguments is for. Yield is really tricky and it's actually a very advanced ruby topic (one which I don't totally understand to be honest) but it's used in this very common place in rails to do a very important thing.